pkg/virtualKubelet/translation/mutation.go

```diff
@@ -180,3 +187,52 @@ func ChangePodIp(newPodCidr string, oldPodIp string) (newPodIp string) {
 	}
 	return strings.TrimSuffix(newPodIpBuilder.String(), ".")
 }
+
+func TranslateSA(destPod *v1.Pod, origPod *v1.Pod, remoteSecrets []interface{}) (*v1.Pod, error) {
+	var secret *v1.Secret = nil
+	for _, obj := range remoteSecrets {
+		s, ok := obj.(*v1.Secret)
+		if !ok {
+			continue
+		}
+		if sa, ok := s.Labels["kubernetes.io/service-account.name"]; ok && sa == origPod.Spec.ServiceAccountName {
```

Comment:
before checking the label, check also that `type==kubernetes.io/service-account-token`